app/core/scheduler/scheduler.py

```diff
@@ -1,84 +1,39 @@
+        self._mode = mode
 
     def __call__(self):
-        signal.signal(signal.SIGINT, signal.SIG_IGN)
-        # ObservatoryProperties.set_properties(GeminiProperties)
-
-        programs = read_ocs_zipfile(os.path.join(ROOT_DIR, 'app', 'data', '2018B_program_samples.zip'))
-
-        # Create the Collector and load the programs.
-        print('Loading programs...')
-        # Parse configs
-        time_slot_length = TimeDelta(self.config.scheduler.time_slot_length * u.min)
-        sites = frozenset(list(map(eval, self.config.scheduler.sites))) \
-            if isinstance(self.config.scheduler.sites, list) else eval(self.config.scheduler.sites)
-        semesters = set(map(eval, self.config.collector.semesters))
-        program_types = set(map(eval, self.config.collector.program_types))
-        obs_classes = set(map(eval, self.config.collector.observation_classes))
-
-        collector = Collector(
-            start_time=self.start_time,
-            end_time=self.end_time,
-            time_slot_length=time_slot_length,
-            sites=sites,
-            semesters=frozenset(semesters),
-            program_types=frozenset(program_types),
-            obs_classes=frozenset(obs_classes)
-        )
-        collector.load_programs(program_provider=OcsProgramProvider(),
-                                data=programs)
-
-        if Site.GS in sites or Site.GN in sites:
-            ObservatoryProperties.set_properties(GeminiProperties)
-        else:
-            raise NotImplementedError('Only Gemini is supported.')
-
-        selector = Selector(collector=collector)
-
-        # Execute the Selector.
-        # Not sure the best way to display the output.
-        selection = selector.select()
-        # Execute the Optimizer.
-        dummy = DummyOptimizer()
-        optimizer = Optimizer(selection, algorithm=dummy)
-        plans = optimizer.schedule()
-
-        # Save to database
-        PlanManager.set_plans(plans)
-        # print(f'{PlanManager._plans=}')
-
-
-def build_scheduler():
-    # TODO: Default values for now but this should be:
-    # start_time = Time(datetime.now(), scale='utc)
-    # end_time = Time(datetime.now() + rest_of_the_night, scale='utc')
-    start = Time("2018-10-01 08:00:00", format='iso', scale='utc')
-    end = Time("2018-10-03 08:00:00", format='iso', scale='utc')
-    return Scheduler(start, end)
+        logging.info(f'Running on mode {self._mode}')
+        self._mode.schedule(self.start_time, self.end_time)
+
+
+def build_scheduler(start: Time = Time("2018-10-01 08:00:00", format='iso', scale='utc'),
+                    end: Time = Time("2018-10-03 08:00:00", format='iso', scale='utc')) -> Scheduler:
+    
+    # Set scheduler mode based on config
+    scheduler = Scheduler(start, end)
+    if config.mode == 'Operation':
+        mode = OperationMode()
+    elif config.mode == 'Validation':
+        mode = ValidationMode()
+    else:
+        mode = SimulationMode()
```

Comment:
Just one comment here: what if it's a mode that doesn't make sense?
Maybe check for Simulation explicitly and raise an exception if it isn't?